File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        for i in range(len(matrix)):
            if matrix[i][-1] < target:
                continue
            else:
                index = bisect.bisect_left(matrix[i], target)
                if 0 <= index <= len(matrix[i]) - 1 and matrix[i][index] == target:
                    return True

        return False

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {

        for (int i = 0; i < matrix.size(); i++) {
            auto lIt = lower_bound(matrix[i].begin(), matrix[i].end(), target);
            if (lIt != matrix[i].end() && *lIt == target) {
                return true;
            }
        }

        return false;
    }
};